static/js/search.js

```diff
@@ -2,9 +2,33 @@ var search = (function () {
 
 var exports = {};
 
+function ValidURL(str)
+{
+    var patt = /(ftp|http|https):\/\/(\w+:{0,1}\w*@)?(\S+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%@!-\/]))?/;
+    if (!patt.test(str)) {
```

Comment:
Would recommend using a more descriptive variable name, e.g. `url_pattern`, for clarity.